lib/autoupdate.ps1

```diff
@@ -431,3 +424,104 @@ function autoupdate([String] $app, $dir, $json, [String] $version, [Hashtable] $
         Write-Host -f DarkGray "No updates for $app"
     }
 }
+
+## Helper Functions
+
+function PropertyHelper {
+    <#
+    .SYNOPSIS
+        Helper of updating property
+    .DESCRIPTION
+        Update manifest property (String, Array or PSCustomObject).
+    .PARAMETER Property
+        Property to be updated
+    .PARAMETER Value
+        New property values
+        Update line by line
+    .OUTPUTS
+        System.Object[]
+            The first element is new property, the second element is change flag
+    #>
+    param (
+        [Object]$Property,
+        [Object]$Value
+    )
+    $hasChanged = $false
+    switch ($Property.GetType().Name) {
+        'String' {
+            $Value = $Value -as [String]
+            if (($null -ne $Value) -and ($value -ne $Property)) {
```

Comment:
What is use case of `$Value -ne $Property`?

The function will be called for example as `Property-Helper $property.Name $property.Value`, where name will be evaluated from manifest as `$manifest.autoupdate.extract_dir`. So name will be `extract_dir` and value could be `mocked v$version`

I do not see a use case when the actual value could be equal to property name itself

Reply:
`$Property` is the property value, or property value's pointer, so we could replace it with newer one (`$Value`).

Reply:
One use case is when nothing changed in update, the manifest will not be updated, even with `-Force`.

Reply:
Manifest have to be updated all the time when Force is given.

Reply:
Even if nothing changed, with only the same file overwrited?

Reply:
Yes. That is how current checkver work.

Reply:
It is used mostly instead of `formatjson`. Since checkver is at the end calling ConvertToPrettyJson

Reply:
This line is useless, I think...

https://github.com/lukesampson/scoop/blob/6eb90c9c112c8c6c1790f1bd4e1c9cd67a6e70df/lib/autoupdate.ps1#L431